### ISSUE TITLE
Support for Buffer.filter(T => ReadChannel[Boolean])

### ISSUE THREAD

**Issue by mkotsbak** [MetaStack-pl/MetaRx#12]
It would be nice to be able to add a filter using a function that returns a channel so that it could be updated when the content changes.

**Comment by tindzk** [comment]
I was thinking about adding these kind of functions myself, but decided against it. It will be hard to implement this efficiently. Another reason is that we will have every combinator twice. Unfortunately, overloading will not work, so we'd have to add a suffix, making the API less intuitive. If you still think this function is a good idea, perhaps you could provide some more context? Often there is another way of structuring your data flow that would not make these kind of functions necessary though.

**Comment by mkotsbak** [comment]
Japp, another way for me is to not filter, but use show() for each element instead.

**Comment by tindzk** [comment]
Yes, that should be quite efficient because you will not have to remove and re-insert any elements when the filtering function changes.
